.github/coding-style.md

```diff
@@ -0,0 +1,171 @@
+		or.SensorOr(user.Authenticated()): {
+			"/v1.MetadataService/GetMetadata",
+		},
+	})
+)
+```
+
+## Guidelines
+
+- Always use `defer mutex.Unlock()` instead of explicitly calling `Unlock()`.
+- A `Close()` on a file etc. should only be deferred and wrapped in
+  `utils.IgnoreError` if you exclusively _read_. If you write to a file, you
+  **must** check for a `nil` error upon close, otherwise you cannot be certain
+  that all data has been persisted.
+- Use the fact that certain operations return zero values by default to write
+  conditions in a more goal-oriented way. E.g., prefer
+  `if x := m[k]; x != nil { foo(*x) }` over `if x, ok := m[k]; ok { foo(*x) }`.
+  The former makes it apparent that the `*x` cannot panic while the latter only
+  after checking that `m` indeed can never contain `nil` values.
+
+### Declarations
+
+- Prefer `var x T` instead of `x := T{}` or `x := 0/false/…` for initializing
+  zero-valued local variables.
+- Avoid `var ( … )` blocks within function scopes, only use them at the global
+  scope.
+- Declarations at the global scope should follow this order: const block, var
+  block, everything else. If you define struct types and methods of this type in
+  the same file, the order should be: type declaration, new/constructor function,
+  methods. Do not interleave methods of different types.
+- If you are registering packages via `_` then extract this line into its own
+  import block and preface it with a comment explaining what this import does.
+- Scope variables as low as possible.
+- Only use `if` + assignment with `:=`.
+
+### Functions
+
+- Every function that can potentially block should receive a `ctx context.Context`
+  as the first parameter.
+- An exported function should only ever return exported types (even though Go
+  permits otherwise).
+- Always pass around proto objects as pointers when traversing function
+  boundaries. If you need to make a copy, use `obj.Clone()`.
+- When using slice tricks like filtering in a function, pass the slice as a
+  pointer to explicitly call out that the underlying data may be modified in
+  that function.
+- Avoid naked returns
+
+### Types and collections
+
+- Use `.GetField()` instead of `.Field` on protobuf objects
+- Do not embed public types that are part of an object’s *internal* state into
+  structs, exported or non-exported. I.e., do `mutex sync.Mutex` instead of
+  `sync.Mutex` at the `struct` level. `obj.Lock()` suggests that Lock would be
+  part of the public interface, which it should not be.
+- When using `append`, preallocate whenever possible.
+- When setting elements in a slice, instantiate like `s := make([]string, 0, size)`
+  and use `append` because this greatly decreases potential index out of bounds
+  errors.
+
+### Error handling
+ 
+- Use `errors.Wrap[f]()` from `github.com/pkg/errors` to add a message when
+  forwarding the error.
+- Use `RoxError.CausedBy[f]()` from `pkg/errox` to add context to an existing
+  message. 
```

Comment:
I think for the `CauseBy` it'd be great to have an example, since it's sometimes not _exactly_ clear what to put there: should it be `RoxError.CausedBy(err)` or `RoxErr.CausedBy(err.Error())` for example?

Reply:
I guess `errox` package has the examples. If they are insufficient I'd still prefer to adjust there for locality.